linux_osl.c no 6.x: skb->users e refcount_t; kernel_read() com a ordem
de argumentos do 4.14+.

--- a/shared/linux_osl.c	2026-10-08 10:09:18.142416343 -0300
+++ b/shared/linux_osl.c	2026-10-08 10:09:18.254419248 -0300
@@ -944,7 +944,7 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 14)
 	skb->list = NULL;
 #endif
-	atomic_set(&skb->users, 1);
+	refcount_set(&skb->users, 1);
 
 	PKTSETCLINK(skb, NULL);
 	PKTCCLRATTR(skb);
@@ -1200,9 +1200,9 @@
 
 #ifdef CTFPOOL
 		if (PKTISFAST(osh, skb)) {
-			if (atomic_read(&skb->users) == 1)
+			if (refcount_read(&skb->users) == 1)
 				smp_rmb();
-			else if (!atomic_dec_and_test(&skb->users))
+			else if (!refcount_dec_and_test(&skb->users))
 				goto next_skb;
 			osl_pktfastfree(osh, skb);
 		} else
@@ -2431,9 +2431,8 @@
 	if (!image)
 		return 0;
 
-	rdlen = kernel_read(fp, fp->f_pos, buf, len);
-	if (rdlen > 0)
-		fp->f_pos += rdlen;
+	/* 4.14+: kernel_read(file, buf, count, &pos) atualiza a posicao */
+	rdlen = kernel_read(fp, buf, len, &fp->f_pos);
 
 	return rdlen;
 }
